bot/messaging/events.py

```diff
@@ -475,5 +475,76 @@ def on_member_ban(self):
         """
         return 'on_member_ban'
 
+    @property
+    def on_semester_archive(self):
+        """
+        Published when a semester is manually archived or scheduled to be archived.
+
+        Args:
+            semester (bot.models.class_models.ClassSemester): The semester to archive.
+            inter (discord.Interaction | None = None): The app command interaction.
+        """
+        return 'on_semester_archive'
```

Comment:
I fucking hate this events class I wish there was a better way of doing this

Reply:
LMAO if you can think of any better way I'll def. consider making a PR for it